contrib/HDRMerge.lua

```diff
@@ -0,0 +1,417 @@
+    soft_min = 1,
+    soft_max = 30,
+    hard_min = 1,
+    hard_max = 3600,
+    step = 1,
+    digits = 0,
+    value = temp,
+    sensitive = GUI.HDR.batch.value,
+    reset_callback = function(self) 
+        self.value = 3
+    end
+}
+local lbl_import = dt.new_widget('section_label'){
+    label = 'import options'
+}
+GUI.Target.style = dt.new_widget('combobox'){
+    label = 'apply style on import',
+    tooltip = "Apply selected style on auto-import to newly created image",
+    selected = 1,
+    "none",
+    changed_callback = function(self)
+        dt.preferences.write(mod, 'active_style', 'string', self.value) 
+        dt.preferences.write(mod, 'active_style_ind', 'integer', self.selected)
+    end,
+    reset_callback = function(self) 
+        self.selected = 1
+        dt.preferences.write(mod, 'active_style', 'string', self.value) 
+        dt.preferences.write(mod, 'active_style_ind', 'integer', self.selected)
+    end 
+}
+for k=1, (styles_count-1) do
+    GUI.Target.style[k+1] = styles[k].name
+end
+temp = dt.preferences.read(mod, 'active_style_ind', 'integer')
+if not InRange(temp, 1, styles_count) then temp = 1 end
+GUI.Target.style.selected = temp
+GUI.Target.copy_tags = dt.new_widget('check_button'){
+    label = 'copy tags',
+    value = dt.preferences.read(mod, 'active_copy_tags', 'bool'),
+    tooltip = 'copy tags from first source image',
+    clicked_callback = function(self) dt.preferences.write(mod, 'active_copy_tags', 'bool', self.value) end,
+    reset_callback = function(self) self.value = true end
+}
+temp = dt.preferences.read(mod, 'active_add_tags', 'string')
+if temp == '' then temp = nil end 
+GUI.Target.add_tags = dt.new_widget("entry"){
+    tooltip = "Additional tags to be added on import. Seperate with commas, all spaces will be removed",
+    text = temp,
+    placeholder = "Enter tags, seperated by commas",
+    editable = true
+}
+GUI.run = dt.new_widget("button"){
+    label = 'merge',
+    tooltip ='run HDRMerge with the above specified settings',
+    clicked_callback = function() main() end
+}
+GUI.exes.HDRMerge = dt.new_widget('file_chooser_button'){
+    title = "Select HDRmerge executable",
+    value = df.get_executable_path_preference(HDRM.name),
+    is_directory = false
+}
+GUI.exes.update = dt.new_widget('button'){
+    label = 'update',
+    tooltip ='update the binary path with current value',
+    clicked_callback = function() ExeUpdate({HDRM}) end
+}
+GUI.options = dt.new_widget('box'){
+    orientation = 'vertical',
+    lbl_hdr,
+    GUI.HDR.bps,
+    GUI.HDR.size,
+    GUI.HDR.batch,
+    GUI.HDR.gap,
+    lbl_import,
+    GUI.Target.style,
+    GUI.Target.copy_tags,
+    GUI.Target.add_tags,
+    GUI.run
+}
+local exes_box = dt.new_widget('box'){
+    orientation = 'vertical',
+    GUI.exes.HDRMerge,
+    GUI.exes.update
+}
+GUI.stack = dt.new_widget('stack'){
+    GUI.options,
+    exes_box
+}
+if dt.preferences.read(mod, 'bin_exists', 'bool') then
+   GUI.stack.active = 1

```

Comment:
Not a real review, but here is a space missing.

Reply:
Added space in latest commit